packages/flutter/lib/src/painting/cupertino_rounded_rectangle_border.dart

```diff
@@ -0,0 +1,504 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+import 'dart:ui' as ui;
+
+import 'basic_types.dart';
+import 'borders.dart';
+import 'edge_insets.dart';
+
+/// The corner mode of a Cupertino Rounded Rectangle.
+///
+/// In the event that the height or width of the rectangle is less than 3x its
+/// radius, either the shape of the rectangle or the radius of each corner can
+/// be changed to accommodate the given rounded rectangle parameters.
+enum RoundedRectCornerMode {
+  /// In an attempt to keep the corner radius of a given rectangle roughly the
+  /// same regardless of dimension, the shape of the rect corners will
+  /// change when the height or width is less than 3x the radius.
+  ///
+  /// This option is best used in scenarios when a rectangle is static and a smooth
+  /// transition between different rect sizes is not necessary.
+  dynamicShape,
+
+  /// In an attempt to keep the shape of the rectangle the same regardless of its
+  /// dimension, the radius will automatically be lessened to maximize the
+  /// roundness of the resulting rectangle if its width or height is less than
+  /// 3x the radius.
+  ///
+  /// This option is best used in scenarios where a rectangle will be animated
+  /// between various different dimensions.
+  dynamicRadius,
+}
+
+/// Creates a Cupertino styled rounded rectangle - a shape similar to a rounded
+/// rectangle, but with a smoother transition from the sides to the rounded
+/// corners.
+///
+/// In this shape, the curvature of each corner over the arc is approximately
+/// a gaussian curve instead of a step function as with a traditional quarter
+/// circle round.
+///
```

Comment:
Mention why it's in Cupertino, that it's an approximation of a superellipse and why it isn't just in widgets.